src/resolvers/Mutation/updateAdvertisement.ts

```diff
@@ -0,0 +1,83 @@
+import type { MutationResolvers } from "../../types/generatedGraphQLTypes";
+import { Advertisement, User } from "../../models";
+import { errors, requestContext } from "../../libraries";
+import {
+  ADVERTISEMENT_NOT_FOUND_ERROR,
+  USER_NOT_FOUND_ERROR,
+  INPUT_NOT_FOUND_ERROR,
+  END_DATE_VALIDATION_ERROR,
+  START_DATE_VALIDATION_ERROR,
+} from "../../constants";
+
+export const updateAdvertisement: MutationResolvers["updateAdvertisement"] =
```

Comment:
who is allowed to execute this mutation? 
1. superadmin
2. superadmin, admin
3. superadmin, admin, user

please confirm this

Reply:
@xoldyckk For now, we'll maintain the authentication level as '@auth'. We'll create a new issue to address this matter. According to our use case, this mutation should be allowed for superadmins, admins, and the user who created the advertisement. However, in our current codebase, we don't keep track of the user who created the advertisement when it's initially created. This absence of tracking makes it challenging to verify whether the user attempting to update the advertisement is the original creator. Is this okay or do we need to do something else?

Reply:
@palisadoes

Reply:
Ads management would be by SuperAdmin and Admin users

Reply:
without a role based authorization logic this PR cannot be approved